aria/src/lib.rs

```diff
@@ -0,0 +1,412 @@
+
+impl KeyInit for Aria256 {
+    fn new(key: &Key<Self>) -> Self {
+        let kl = u128::from_be_bytes(key[0..16].try_into().unwrap());
+        let kr = u128::from_be_bytes(key[16..32].try_into().unwrap());
+
+        let w0 = kl;
+        let w1 = fo(w0 ^ C3) ^ kr;
+        let w2 = fe(w1 ^ C1) ^ w0;
+        let w3 = fo(w2 ^ C2) ^ w1;
+
+        let ek = [
+            w0 ^ w1.rotate_right(19),
+            w1 ^ w2.rotate_right(19),
+            w2 ^ w3.rotate_right(19),
+            w3 ^ w0.rotate_right(19),
+            w0 ^ w1.rotate_right(31),
+            w1 ^ w2.rotate_right(31),
+            w2 ^ w3.rotate_right(31),
+            w3 ^ w0.rotate_right(31),
+            w0 ^ w1.rotate_left(61),
+            w1 ^ w2.rotate_left(61),
+            w2 ^ w3.rotate_left(61),
+            w3 ^ w0.rotate_left(61),
+            w0 ^ w1.rotate_left(31),
+            w1 ^ w2.rotate_left(31),
+            w2 ^ w3.rotate_left(31),
+            w3 ^ w0.rotate_left(31),
+            w0 ^ w1.rotate_left(19),
+        ];
+
+        let dk = [
+            ek[16],
+            a(ek[15]),
+            a(ek[14]),
+            a(ek[13]),
+            a(ek[12]),
+            a(ek[11]),
+            a(ek[10]),
+            a(ek[9]),
+            a(ek[8]),
+            a(ek[7]),
+            a(ek[6]),
+            a(ek[5]),
+            a(ek[4]),
+            a(ek[3]),
+            a(ek[2]),
+            a(ek[1]),
+            ek[0],
+        ];
+
+        Self { ek, dk }
+    }
+}
+
+macro_rules! impl_aria {
+    ($name:ident, $subkey_size:literal, $key_size:ty, $doc:literal) => {
+        #[doc = $doc]
+        #[derive(Clone)]
+        pub struct $name {
+            /// Encrypting subkeys.
+            ek: [u128; $subkey_size],
+            /// Encrypting subkeys.
+            dk: [u128; $subkey_size],
+        }
+        impl BlockCipher for $name {}
+
+        impl KeySizeUser for $name {
+            type KeySize = $key_size;
+        }
+
+        impl fmt::Debug for $name {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                f.write_str(concat!(stringify!($name), " { ... }"))
+            }
+        }
+
+        impl AlgorithmName for $name {
+            fn write_alg_name(f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                f.write_str(stringify!($name))
+            }
+        }
+
+        #[cfg(feature = "zeroize")]
+        #[cfg_attr(docsrs, doc(cfg(feature = "zeroize")))]
+        impl Drop for $name {
+            fn drop(&mut self) {
+                self.ek.zeroize();
+                self.dk.zeroize();
+            }
+        }
+
+        #[cfg(feature = "zeroize")]
+        #[cfg_attr(docsrs, doc(cfg(feature = "zeroize")))]
+        impl ZeroizeOnDrop for $name {}
+
+        cipher::impl_simple_block_encdec!(
+            $name, U16, cipher, block,
+            encrypt: {
+                let b = block.get_in();
+
+                let mut p0 = u128::from_be_bytes(b[0..16].try_into().unwrap());
```

Comment:
`b` has type `&GenericArray<u8, U16>`, so you should be able to use `Into` for converting it into `[u8; 16].`